qtism/runtime/expressions/MapResponseProcessor.php

```diff
@@ -97,14 +98,25 @@ public function process()
                         }
                         
                         if ($val instanceof Comparable && $val->equals($mapKey) || $val === $mapKey) {
-                            return new QtiFloat($mapEntry->getMappedValue());
+                            $result = $mapEntry->getMappedValue();
                         } elseif ($variable->getBaseType() === BaseType::STRING && $val === null && $mapKey === '') {
-                            return new QtiFloat($mapEntry->getMappedValue());
+                            $result = $mapEntry->getMappedValue();
```

Comment:
Maybe should you break the loop after having found what you were searching for.
Or maybe are you expecting to find the last matching entry?